[Stuart Long] iP

s7u4rt99 opened the pull request:
### Duke

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here is the `main` method:
````java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

Commit:
Add Gradle support

s7u4rt99 committed:
Complete Level 1

s7u4rt99 committed:
Complete Level 2

s7u4rt99 committed:
Complete Level 3

s7u4rt99 committed:
Complete Level 4

s7u4rt99 committed:
Complete A-TextUiTesting

s7u4rt99 committed:
Complete Level-5

s7u4rt99 committed:
Complete Level-6

s7u4rt99 committed:
Complete Level-7

s7u4rt99 committed:
Complete Level-8

s7u4rt99 committed:
/Users/stuartlong/Desktop/CS2103T/ip Merge branch 'branch-Level-7'
Complete Level-7

s7u4rt99 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

s7u4rt99 committed:
Complete A-MoreOOP

s7u4rt99 committed:
Complete A-Packages

s7u4rt99 committed:
Complete A-JUnit

s7u4rt99 committed:
Enable OS dependent file saving

s7u4rt99 committed:
Complete A-JavaDoc

s7u4rt99 committed:
Complete A-CodingStandard.

s7u4rt99 committed:
Fix JavaDocs

s7u4rt99 committed:
Complete Level-9

s7u4rt99 committed:
Merge branch 'branch-A-JavaDoc'

s7u4rt99 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/RemoveCommand.java
#	src/main/java/duke/exception/IncompleteDeadlineException.java
#	src/main/java/duke/exception/IncompleteEventException.java
#	src/main/java/duke/exception/IncompleteToDoException.java
#	src/main/java/duke/exception/InvalidCommandException.java
#	src/main/java/duke/task/ToDo.java

s7u4rt99 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/task/Task.java

pss-coder commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+import duke.exception.DukeException;
+
+/**
+ * Represents a program called Duke. A <code> Duke </code> is a software which
```
I like the use of the code tag to contrast the program's name.

pss-coder commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+public class Duke {
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList tasks;
```
I like the idea of the TaskList class, perhaps the name of the variable could be taskList instead ?

pss-coder commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,223 @@
+     * @return String[] containing the description at index 0 and the deadline (if any) at index 1.
+     * @throws DukeException If no deadline or description for respective Task objects.
+     */
+    public static String[] splitInput(String input, String type) throws DukeException {
```
I like how the logic of the code was group neatly. Should there also be comments to better explain what the groups of code is trying to achieve ?

pss-coder commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,118 @@
+ * Stores and retrieves information of the tasklist for Duke.
+ */
+public class Storage {
+    private ArrayList<Task> toStore;
```
Could the variable name be stored tasksToStore instead ? or tasks ?

pss-coder commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+     * @param index Index of Task to get the String representation.
+     * @return String representation of Task in tasks with given index.
+     */
+    public String taskToString(int index) {
```
Would perhaps starting with a verb such as ConvertTaskToString() would align with the convention more ?

pss-coder commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+     * @param index Index of Task to get the String representation to save.
+     * @return String representation of Task in tasks with given index.
+     */
+    public String taskSaveToString(int index) {
```
Similarly, how about SaveTaskToString() instead sound ?

pss-coder commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,124 @@
+ */
+public class Ui {
+    public static String LINE = "____________________________________________________________";
+    private final Scanner sc;
```
As this is a constant variable, shouldn't the variable name be all uppercase ?

pss-coder commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,124 @@
+ * Represents the Ui of Duke.
+ */
+public class Ui {
+    public static String LINE = "____________________________________________________________";
```
I like the use of abstracting out the formatting. Also, shouldn't a final keyword be in here, give the all caps variable name ?

s7u4rt99 replied:
Hi @pss-coder! Thank you for reviewing my PR and giving me valuable feedback, I did not realise that I had so many naming conventions error, I will go and fix them. Thanks once again.

pss-coder reviewed (commented):
> Overall: LGTM, just a couple of naming convention nits to fix. 👍🏽 

Hi Stuart, I was assigned to review your PR. I really liked your use of abstraction with the Commands. I never really taught of using that way. It really inspired me to rework on my code. 😄

s7u4rt99 committed:
Fix IO testing

s7u4rt99 committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branch-A-Gradle

s7u4rt99 committed:
Complete A-Gradle

s7u4rt99 committed:
Merge branch 'branch-A-Gradle'

s7u4rt99 committed:
Complete A-Checkstyle

s7u4rt99 committed:
Merge branch 'branch-A-Checkstyle'

s7u4rt99 committed:
Complete Level-10

s7u4rt99 committed:
Merge branch 'branch-Level-10'

s7u4rt99 committed:
Fix coding style

s7u4rt99 committed:
Change mainClassName

s7u4rt99 committed:
Complete A-Assertions

Classes do not have any checks for their invariants
or assumptions made.

Not checking for these invariants or assumptions
will cause the program to crash or give incorrect
results if these invariants or assumptions are not
met.

Add assertions to check for invariants in the code.

Assertions is preferred as it gets the job done and it
can be turned on or off.

s7u4rt99 committed:
Merge pull request #1 from s7u4rt99/branch-A-Assertions

Complete A-Assertions

s7u4rt99 committed:
Complete A-CodeQuality

Code Quality can be improved, after making lots of changes with the
code.

Poor code quality can lead to the code being very messy and hard
to understand.

Follow coding standard of CS2103T to standardise the coding
styles.

Others who read my code would be able to understand what
I am trying to do.

s7u4rt99 committed:
Merge pull request #2 from s7u4rt99/branch-A-CodeQuality

Complete A-CodeQuality

s7u4rt99 committed:
Complete A-Assertions

Classes do not have any checks for their invariants
or assumptions made.

Not checking for these invariants or assumptions
will cause the program to crash or give incorrect
results if these invariants or assumptions are not
met.

Add assertions to check for invariants in the code.

Assertions is preferred as it gets the job done and it
can be turned on or off.

s7u4rt99 committed:
Complete A-CodeQuality

Code Quality can be improved, after making lots of changes with the
code.

Poor code quality can lead to the code being very messy and hard
to understand.

Follow coding standard of CS2103T to standardise the coding
styles.

Others who read my code would be able to understand what
I am trying to do.

s7u4rt99 committed:
Change namings of format methods

s7u4rt99 committed:
Add remove and done all feature

s7u4rt99 committed:
Add archive feature

s7u4rt99 committed:
Add archive all feature

s7u4rt99 committed:
Add remove archive feature and javadocs

s7u4rt99 committed:
Fix bugs from and finalised archive method

No way to archive methods now, we can only delete
items if we do not want them, and there is no way to save them.

Without archive method, people are not able to store their deleted
tasks, and it may contain important information.

Create an abstraction to manage archiving of information
and deleting from archive when not needed anymore.

We call the class ArchiveCommand which managed the archiving
of tasks and stores them in the ArchiveList class.

s7u4rt99 committed:
Merge pull request #8 from s7u4rt99/branch-C-Archive

Complete C-Archive

s7u4rt99 committed:
Merge branch 'master' into branch-A-CodeQuality2

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/FindCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/RemoveCommand.java

s7u4rt99 committed:
Merge branch 'master' into branch-A-Assertions2

# Conflicts:
#	src/main/java/duke/Parser.java

s7u4rt99 committed:
Merge pull request #7 from s7u4rt99/branch-A-CodeQuality2

Complete A-CodeQuality

s7u4rt99 committed:
Merge branch 'master' into branch-A-Assertions2

s7u4rt99 committed:
Merge pull request #6 from s7u4rt99/branch-A-Assertions2

Complete A-Assertions

s7u4rt99 committed:
Add new exceptions

s7u4rt99 committed:
Add unarchive feature

s7u4rt99 committed:
Fix javadocs

s7u4rt99 committed:
Align to coding conventions

s7u4rt99 committed:
Add checkstyle to github

s7u4rt99 committed:
Fix magic number

s7u4rt99 committed:
Complete A-BetterGui

s7u4rt99 committed:
Add product website

s7u4rt99 committed:
Add User Guide

s7u4rt99 committed:
Fix user guide

s7u4rt99 committed:
Add photo to user guide

s7u4rt99 committed:
Edit photo size

s7u4rt99 committed:
Resize photos for user guide

s7u4rt99 committed:
Remove unnecessary photos for user guide

s7u4rt99 committed:
FIx coding style

s7u4rt99 committed:
Refactor parser class

s7u4rt99 committed:
Fix javadocs

s7u4rt99 committed:
Fix generateCommand for each Command

s7u4rt99 committed:
Fix bug where it does not save during crash

s7u4rt99 committed:
Change remove command to delete

s7u4rt99 committed:
Fix javadocs

s7u4rt99 committed:
Fix Javadocs

s7u4rt99 committed:
Update user guide

s7u4rt99 committed:
Add javadoc for instance and static variables

s7u4rt99 committed:
Improve find command to find partial words
